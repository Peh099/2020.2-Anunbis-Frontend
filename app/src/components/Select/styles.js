import styled from 'styled-components';

export const SelectStyle = styled.select`
    display: inline-block;
    border: none;
    border-bottom: 1px solid #000000;
    background: ${props => props.backColor ? props.backColor : "white"};
    height: inherit;
    align-self: right;
    width: ${props => props.width ? props.width : "inherit"};
    color: rgba(61, 58, 58, 0.603);
`;