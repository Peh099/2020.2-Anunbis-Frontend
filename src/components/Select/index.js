import React from "react";

export default function Select(props){
    return(
        <div className="Select">
            <select id={props.id} name={props.name} ref={props.register}>
                <option value="">Selecione o seu curso</option>
                {props.options.map((item) => (
                    <option value={item.id_course} key={item.id_course}>{item.name}</option>
                ))}
            </select>
        </div>
    );
}