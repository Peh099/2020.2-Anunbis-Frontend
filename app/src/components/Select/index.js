/* eslint-disable */
import React from 'react';
import { SelectStyle } from './styles';

export default function Select({
  text,
  id,
  name,
  register,
  options,
  backColor,
  width,
  error,
  onChange,
  display,
  testid,
}) {
  return (
    <SelectStyle
      id={id}
      name={name}
      ref={register}
      backColor={backColor}
      width={width}
      error={error}
      onChange={onChange}
      display={display}
      data-testid={testid}
    >
      <option value="" disabled selected>
        {text}
      </option>
      {options?.map((item) => (
        <option
          value={item.id}
          key={item.id}
          selected={item.selected ? item.selected : false}
        >
          {item.name}
        </option>
      ))}
    </SelectStyle>
  );
}
