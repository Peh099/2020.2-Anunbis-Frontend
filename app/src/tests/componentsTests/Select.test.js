import renderer from 'react-test-renderer';
import Select from '../../components/Select';

describe('Snapshot Select component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Select />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
