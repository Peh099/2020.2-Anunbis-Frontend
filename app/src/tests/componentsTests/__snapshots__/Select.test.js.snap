// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Select component matches the snapshot 1`] = `
<select
  className="sc-bdnxRM gkBzG"
>
  <option
    disabled={true}
    selected={true}
    value=""
  />
</select>
`;
